src/Symfony/Component/Messenger/Tests/Debug/LoggingMiddlewareTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Tests\Debug;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Debug\LoggingMiddleware;
+use Symfony\Component\Messenger\Tests\Fixtures\DummyMessage;
+
+class LoggingMiddlewareTest extends TestCase
+{
+    public function testDebugLogAndNextMiddleware()
+    {
+        $message = new DummyMessage('Hey');
+
+        $logger = $this->createMock(LoggerInterface::class);
+        $logger->expects($this->exactly(2))->method('debug');
+
+        $next = $this->createPartialMock(\stdClass::class, array('__invoke'));
+        $next->expects($this->once())->method('__invoke')->with($message);
+
+        (new LoggingMiddleware($logger))->handle($message, $next);
```

Comment:
Can you test that it returns the value returned by the previous middleware as well?

Reply:
Done.